kedro/pipeline/pipeline.py

```diff
@@ -369,6 +369,31 @@ def grouped_nodes(self) -> list[list[Node]]:
 
         return [list(group) for group in self._toposorted_groups]
 
+    @property
+    def grouped_nodes_by_namespace(self) -> dict[str, dict[str, Any]]:
+        """Return a dictionary of the pipeline nodes grouped by namespace with
```

Comment:
I think it would be helpful to add to the docstring what the structure of the returned dictionary is, maybe by providing an example. Probably also helpful to add what the possible types are.